### ISSUE TITLE
[ Bug ] widget name missing in COMBO widgets [ make it a feature ]

### ISSUE THREAD

**Issue by JorgeR81** [chrisgoringe/cg-controller#450]
The widget name missing in COMBO widgets

![x1](https://github.com/user-attachments/assets/619c1abe-1d26-48ed-a3c9-3a25ba27d080)

----

But I actually like this !

Can you make this an option, per node, via right click.

We could use it in the "Load" nodes.
This way, the long model names would fit better.
And they are also easier to read, when aligned to the right. 

![x4](https://github.com/user-attachments/assets/1d7c661a-da9b-40cd-92bc-af293adc8df6)

![x3](https://github.com/user-attachments/assets/ebeace0f-7d45-48ce-b3a4-23dc2348a882)

**Comment by JorgeR81** [comment]
This it's not only a feature idea.

It's also a Bug. 
Now we can't see the names of the combo widgtes. But, before we could. 

See first image: 

I can see `karras`, but not `scheduler`.

----

Maybe they should be separate issues ?

So that you can fix the Bug now, and add the Feature later ?

**Comment by chrisgoringe** [comment]
bug done on dev. I'm not sure about the feature, but happy for someone to raise it separately!

**Comment by JorgeR81** [comment]
About the "feature" part, it could also be integrated in `Custom names for widgets` ( https://github.com/chrisgoringe/cg-controller/issues/386 ), if this is added.

We could allow deleting all characters, when changing the widget's name.
And in this case, the text of the combo list ( e.g. model name ) would be aligned to the right.

**Comment by JorgeR81** [comment]
![log](https://github.com/user-attachments/assets/e076df10-5922-4ce0-8e06-6475ce8f5805)

**Comment by chrisgoringe** [comment]
fixed, I think

**Comment by JorgeR81** [comment]
It's working, but the combo list text it's still aligned to the left.

![ks](https://github.com/user-attachments/assets/b259e84e-2608-443d-b796-f0020eb0aaca)


----

Also, the combo box it's not the native one, but it does not have subfolders, or a search field.

Is that expected for this version ?

![sx1](https://github.com/user-attachments/assets/5310a030-263e-4256-802d-05f5f6bda523)


![sx](https://github.com/user-attachments/assets/8b3db40d-7c2e-45e1-b93d-16fa607b34f0)

**Comment by chrisgoringe** [comment]
can you check you have the latest dev? I did a couple of check ins, I think you might not have them all.

**Comment by JorgeR81** [comment]
OK, I have updated.

The 2 issues above were fixed. 

But, now the shortening of the Combo widget names, it's starting too early ( e.g. "sampler_name" )

![i3](https://github.com/user-attachments/assets/08224867-ae49-40a4-8877-a41aca482bb6)

![i2](https://github.com/user-attachments/assets/024f12f9-8173-4b31-9889-c654d968fac3)

**Comment by chrisgoringe** [comment]
yeah, there's a better way to do that, but I'll need to do it as part of 1.7 to avoid version control problems with other changes.

**Comment by JorgeR81** [comment]
By the way, in `1.7`, the text it's still aligned to the left.

**Comment by JorgeR81** [comment]
> By the way, in 1.7, the text it's still aligned to the left.

This is now fixed.
